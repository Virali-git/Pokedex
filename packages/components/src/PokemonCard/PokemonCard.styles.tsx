export const CardStyle = {
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
  borderRadius: "12px",
  overflow: "hidden",
};
export const Boxstyle = {
  width: "100%",
  height: "200px",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  backgroundColor: "#f3f3f3",
};

export const CardMediaStyle = {
  width: "100%",
  height: "300px",
  objectFit: "cover",
  borderTopLeftRadius: "12px",
  borderTopRightRadius: "12px",
};
export const CardContentTypoStyle = {
  textTransform: "capitalize",
  fontWeight: "bold",
  color: "#333",
};
export const CardContentStyle = { textAlign: "center" };
export const ListItemStyle = { lineHeight: "1px", textAlign: "left" };
export const TypoBoxStyle = { fontWeight: "bold", color: "#333" };
export const bottomTextStyle = {
  display: "flex",
  gap: 5,
  justifyContent: "center",
};
export const LinkStyle = {
  textDecoration: "none",
  color: "#007bff",
  fontWeight: "bold",
};
export const BoxMarginStyle = { marginTop: "20px" };
export const ColumnTypoStyle = {
  display: "flex",
  flexDirection: "column",
  marginTop: "10px",
};
