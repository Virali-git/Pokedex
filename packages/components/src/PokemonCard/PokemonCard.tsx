import React, { useState } from "react";
import {
  Card,
  CardContent,
  CardMedia,
  CircularProgress,
  Box,
  Typography,
  List,
  ListItem,
  ListItemText,
} from "@mui/material";

 type AbilitiesType = {
    ability: {
      name: string;
      url: string;
    };
  };
 interface CardType {
    name: string;
    height?: number;
    weight?: number;
    location?: string;
    abilities?: AbilitiesType[];
    image?:string;
  }

export const PokemonCard  = (props: CardType) => {
  const [loading, setLoading] = React.useState(true);

  return (
    <Card
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
        borderRadius: "12px",
        overflow: "hidden",
      }}
    >
      {loading && (
        <Box
          sx={{
            width: "100%",
            height: "200px",
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
            backgroundColor: "#f3f3f3",
          }}
        >
          <CircularProgress color="secondary" />
        </Box>
      )}

      <CardMedia
        sx={{
          width: "100%",
          height: "300px",
          objectFit: "cover",
          borderTopLeftRadius: "12px",
          borderTopRightRadius: "12px",
        }}
        component="img"
        image={props.image}
        title={props.name}
        loading="lazy"
        onLoad={() => setLoading(false)}
        onError={() => setLoading(false)}
      />
      <CardContent sx={{ textAlign: "center" }}>
        <Typography
          gutterBottom
          variant="h4"
          component="div"
          sx={{
            textTransform: "capitalize",
            fontWeight: "bold",
            color: "#333",
          }}
        >
          {props.name}
        </Typography>
        <Box sx={{ display: "flex", gap: 5, justifyContent: "center" }}>
          <Typography variant="body2" color="text.secondary">
            Height: {props.height}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            Weight: {props.weight}
          </Typography>
        </Box>
        <Typography
          variant="body2"
          color="text.secondary"
          sx={{ display: "flex", flexDirection: "column", marginTop: "10px" }}
        >
          Location:
          <a
            href={props.location}
            target="_blank"
            rel="noreferrer"
            style={{
              textDecoration: "none",
              color: "#007bff",
              fontWeight: "bold",
            }}
          >
            {props.location}
          </a>
        </Typography>
        <Box sx={{ marginTop: "20px" }}>
          <Typography
            variant="h6"
            component="div"
            sx={{ fontWeight: "bold", color: "#333" }}
          >
            Abilities
          </Typography>
          <List dense={false} sx={{ lineHeight: "1px", textAlign: "left" }}>
            {props.abilities.map((ability, i) => (
              <ListItem key={i}>
                <ListItemText
                  primary={ability.ability.name}
                  secondary={ability.ability.url}
                />
              </ListItem>
            ))}
          </List>
        </Box>
      </CardContent>
    </Card>
  );
};

