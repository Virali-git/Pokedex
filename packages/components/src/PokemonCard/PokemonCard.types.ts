export type AbilitiesType = {
  ability: {
    name: string;
    url: string;
  };
};
export interface CardType {
  name: string;
  height?: number;
  weight?: number;
  location?: string;
  abilities?: AbilitiesType[];
  image?: string;
}
