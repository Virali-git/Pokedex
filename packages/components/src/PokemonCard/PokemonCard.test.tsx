import React from "react";
import { render } from "@testing-library/react";
import { PokemonCard } from "./PokemonCard";

const mockCardData = {
  name: "Pikachu",
  height: 40,
  weight: 60,
  location: "https://example.com/pikachu",
  abilities: [
    {
      ability: {
        name: "Static",
        url: "https://example.com/ability/static",
      },
    },
    {
      ability: {
        name: "Lightning Rod",
        url: "https://example.com/ability/lightning-rod",
      },
    },
  ],
  image: "https://example.com/pikachu-image.jpg",
};

describe("PokemonCard component", () => {
  test("renders Pokemon name", () => {
    const { getByText } = render(<PokemonCard {...mockCardData} />);
    const nameElement = getByText("Pikachu");
    expect(nameElement).toBeInTheDocument();
  });

  test("renders Pokemon height and weight", () => {
    const { getByText } = render(<PokemonCard {...mockCardData} />);
    const heightElement = getByText("Height: 40");
    const weightElement = getByText("Weight: 60");
    expect(heightElement).toBeInTheDocument();
    expect(weightElement).toBeInTheDocument();
  });

  test("renders Pokemon location link", () => {
    const { getByText } = render(<PokemonCard {...mockCardData} />);
    const locationLinkElement = getByText("https://example.com/pikachu");
    expect(locationLinkElement).toBeInTheDocument();
    expect(locationLinkElement).toHaveAttribute(
      "href",
      "https://example.com/pikachu"
    );
  });

  test("renders Pokemon abilities", () => {
    const { getByText } = render(<PokemonCard {...mockCardData} />);
    const ability1Element = getByText("Static");
    const ability2Element = getByText("Lightning Rod");
    expect(ability1Element).toBeInTheDocument();
    expect(ability2Element).toBeInTheDocument();
    expect(getByText).toMatchSnapshot();
  });
});
