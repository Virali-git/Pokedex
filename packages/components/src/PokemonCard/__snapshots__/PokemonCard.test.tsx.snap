// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard component renders Pokemon abilities 1`] = `[Function]`;
